indexer/cmd/main.go

```diff
@@ -0,0 +1,159 @@
+				Value:   "http://localhost:8545",
+				Usage:   "Ethereum node endpoint",
+			},
+			&cli.StringFlag{
+				Name:    "elasticsearch-endpoint",
+				EnvVars: []string{"INDEXER_ELASTICSEARCH_ENDPOINT"},
+				Value:   "http://localhost:9200",
+				Usage:   "Elasticsearch endpoint",
+			},
+			&cli.StringFlag{
+				Name:    "es-username",
+				EnvVars: []string{"INDEXER_ES_USERNAME"},
+				Value:   "",
+				Usage:   "Elasticsearch username",
+			},
+			&cli.StringFlag{
+				Name:    "es-password",
+				EnvVars: []string{"INDEXER_ES_PASSWORD"},
+				Value:   "",
+				Usage:   "Elasticsearch password",
+			},
+			&cli.DurationFlag{
+				Name:    "index-interval",
+				EnvVars: []string{"INDEXER_INDEX_INTERVAL"},
+				Value:   15 * time.Second,
+				Usage:   "Interval between indexing operations",
+			},
+			&cli.StringFlag{
+				Name:    "log-level",
+				EnvVars: []string{"INDEXER_LOG_LEVEL"},
+				Value:   "info",
+				Usage:   "Log level (debug, info, warn, error)",
+			},
+			&cli.StringFlag{
+				Name:    "log-fmt",
+				Usage:   "log format to use, options are 'text' or 'json'",
+				EnvVars: []string{"INDEXER_LOG_FMT"},
+				Value:   "text",
+				Action: func(ctx *cli.Context, v string) error {
+					if v != "text" && v != "json" {
+						return fmt.Errorf("invalid log format: %s. Must be 'text' or 'json'", v)
+					}
+					return nil
+				},
+			},
+			&cli.StringFlag{
+				Name:    "log-tags",
+				Usage:   "log tags is a comma-separated list of <name:value> pairs that will be inserted into each log line",
+				EnvVars: []string{"INDEXER_LOG_TAGS"},
+				Action: func(ctx *cli.Context, s string) error {
+					for i, p := range strings.Split(s, ",") {
+						if len(strings.Split(p, ":")) != 2 {
+							return fmt.Errorf("invalid log-tags at index %d, expecting <name:value>", i)
+						}
+					}
+					return nil
+				},
+			},
+		},
+		Action: run,
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func parseLogTags(tagString string) map[string]string {
+	tags := make(map[string]string)
+	for _, p := range strings.Split(tagString, ",") {
+		parts := strings.Split(p, ":")
+		if len(parts) == 2 {
+			tags[parts[0]] = parts[1]
+		}
+	}
+	return tags
+}
+
+func run(cliCtx *cli.Context) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ethClient, err := ethclient.NewW3EthereumClient(cliCtx.String("ethereum-endpoint"))
+	if err != nil {
+		slog.Error("failed to create Ethereum client", "error", err)
+		return err
+	}
+
+	esClient, err := store.NewESClient(cliCtx.String("elasticsearch-endpoint"), cliCtx.String("es-username"), cliCtx.String("es-password"))
+	if err != nil {
+		slog.Error("failed to create Elasticsearch client", "error", err)
+		return err
+	}
+	defer func() {
+		if err := esClient.Close(ctx); err != nil {
+			slog.Error("Failed to close Elasticsearch client", "error", err)
+		}
+	}()
+
+	var esStorage store.Storage = esClient
```

Comment:
The `esClient` can be passed to `NewBlockchainIndexer` without creating a new variable.